### PR TITLE
[DO NOT MERGE] fix: remove instant-finality hack

### PR DESCRIPTION
**Pull request by RetricSu**
todo: support pending for transactions / logs

### CODE REVIEW

File: packages/api-server/src/methods/modules/eth.ts [hunk 1]
```diff
@@ -1147,10 +1108,9 @@ export class Eth {
   ): Promise<GodwokenBlockParameter> {
     switch (blockParameter) {
       case "latest":
-        return undefined;
+        return await this.getTipNumber();
```
Comment (keroro520) [inline packages/api-server/src/methods/modules/eth.ts:1111]: Is it a bugfix that is out of this PR?

**Review by keroro520 — APPROVED** [review]

File: packages/api-server/src/methods/modules/eth.ts (hunk 1 above)
Comment (RetricSu) [reply packages/api-server/src/methods/modules/eth.ts:1111]: no, undefined means mempool in godwoken

**Comment by RetricSu** [comment]
code should be ready. however, after removing the instant-finality feature, we have to tune godwoken block interval to a much short time to enable the same UX for developers. and this depends on godwoken pr to refactor the block producer. we should wait for the upstream updates.

**Comment by RetricSu** [comment]
replaced by 
- https://github.com/godwokenrises/godwoken-web3/pull/568

**Pull request closed without merging**